modules/module-mongodb/src/replication/ChangeStream.ts

```diff
@@ -589,6 +595,11 @@ export class ChangeStream {
 
           const originalChangeDocument = await stream.tryNext();
 
+          // The stream was closed, we will only ever receive `null` from it
+          if (!originalChangeDocument && stream.closed) {
```

Comment:
I noticed this behaviour when testing dropping a MongoDB database while the PowerSync service was actively replicating. The `tryNext` call would emit `null` in an endless loop. We typically receive a `null` response about every second in normal operations, the frequency is much higher after a DB has been dropped.